beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -699,13 +827,13 @@ func (f *blocksFetcher) waitForBandwidth(pid peer.ID, count uint64) error {
 }
 
 func (f *blocksFetcher) hasSufficientBandwidth(peers []peer.ID, count uint64) []peer.ID {
-	filteredPeers := []peer.ID{}
-	for _, p := range peers {
-		if uint64(f.rateLimiter.Remaining(p.String())) < count {
+	filteredPeers := make([]peer.ID, 0, len(peers))
+
+	for _, peer := range peers {
+		if uint64(f.rateLimiter.Remaining(peer.String())) < count {
```

Comment:
hmm why is Remaining function returning an int instead of uint, can remaining be negative?

Reply:
Fixed in bd17dfefb9773bf2778157b8878088b902e62907.

Reply:
actually i see we do this if f.rateLimiter.Remaining(pid.String()) < int64(req.Count) { in several parts of the code lol should we do that or clamp it

Reply:
Added a task in the peerDAS PR. (Normally all occurrences in this PR are clamped.)